vms/evm/database/database_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"bytes"
+	"slices"
+	"testing"
+
+	"github.com/ava-labs/libevm/ethdb"
+	"github.com/ava-labs/libevm/ethdb/dbtest"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/database/memdb"
+
+	avalanchegodb "github.com/ava-labs/avalanchego/database"
+)
+
+// testDatabase wraps the production database with test-only snapshot functionality
+type testDatabase struct {
+	database
```

Comment:
Why do we wrap the `database` type? I think if we wrap the `eth.KeyValueStore` interface we can avoid the cast we do later.